src/cogs/reaction_roles.py

```diff
@@ -0,0 +1,387 @@
+                members_roles: All roles the user currently has. list of role IDs
+                all_relevant_roles: The role IDs that are relevant in this role gorup
+
+            Returns:
+                member.add_roles
+                interaction.response.send_message
+
+            """
+            if str(selected_role) not in str(members_roles):
+                # If the role is NOT in the users roles, add it, and
+                # remove the other related roles if they exist.
+                removals = ''
+                await member.add_roles(target_role)
+                logger.info(f'Added {selection} to {member}.')
+
+                for role in all_relevant_roles:
+                    if str(role) in str(members_roles):
+                        await member.remove_roles(discord.utils.get(member.guild.roles, id=role))
+                        logger.info(f'Removed {selection} from {member}.')
+                        removals = f"{removals}\n - Removed: <@&{role}>"
+
+                await interaction.response.send_message(
+                    f" - Added role: <@&{selected_role}>{removals}"
+                    , ephemeral=True
+                )
+
+        if selection == 'north america':
+            selected_role = fun_roles[selection.replace(" ", "_")]
+            await remove_role_if_exists(
+                selected_role, north_america, members_roles)
+            await add_role_and_remove_others(
+                selected_role, north_america, members_roles, relevant_roles)
+        elif selection == 'europe':
+            await remove_role_if_exists(
+                fun_roles[selection], europe, members_roles)
+            await add_role_and_remove_others(
+                fun_roles[selection], europe, members_roles, relevant_roles)
+        elif selection == 'asia':
+            await remove_role_if_exists(
+                fun_roles[selection], asia, members_roles)
+            await add_role_and_remove_others(
+                fun_roles[selection], asia, members_roles, relevant_roles)
+        elif selection == 'africa':
+            await remove_role_if_exists(
+                fun_roles[selection], africa, members_roles)
+            await add_role_and_remove_others(
+                fun_roles[selection], africa, members_roles, relevant_roles)
+        elif selection == 'south america':
+            selected_role = fun_roles[selection.replace(" ", "_")]
+            await remove_role_if_exists(
+                selected_role, south_america, members_roles)
+            await add_role_and_remove_others(
+                selected_role, south_america, members_roles, relevant_roles)
+        elif selection == 'oceana':
+            await remove_role_if_exists(
+                fun_roles[selection], oceana, members_roles)
+            await add_role_and_remove_others(
+                fun_roles[selection], oceana, members_roles, relevant_roles)
+
+
+class EmploymentRoles(discord.ui.Select):
```

Comment:
All these classes look pretty similar, wondering if we could do a single class to model a Role, and then just create multiple instances of this class passing it config values which are then fed to SelectView's add_tem command in a list comprehension.

Doing this should allow us to easily add /update_role, /add_role and /remove_role commands to allow config directly from discord